protos/flyteidl/core/tasks.proto

```diff
@@ -143,6 +143,15 @@ message TaskTemplate {
     // to use as required.
     // reserve the field numbers 1 through 15 for very frequently occurring message elements
     map<string, string> config = 16;
+
+    // pod_template provides the definition for a PodTemplate to be applied as the base configuration for this task.
+    // This enables fine-grained k8s configuration for Flyte tasks in addition to overriding the default PodTemplate.
+    oneof pod_template {
+        // pod_template_name is the unique name of a PodTemplate k8s resource.
+        string pod_template_name = 19;
+        // pod_template_struct is a JSON marshaled PodTemplate.
```

Comment:
Can you elaborate here on how overriding properties will work? Will service account be copied from the task instead.. how about container images... Etc. Etc.

Reply:
So we have put together pretty [extensive docs on configuring Flyte pods](https://docs.flyte.org/en/latest/deployment/cluster_config/general.html#configuring-k8s-pod) which highlights using the default PodTemplate vs k8s plugin configuration. I imagined we might extend on this (maybe make a separate page?) to try and sort this out end to end. How in-depth do you think is necessary here? We could fill 100 lines I imagine.